enum Protocol {
    UNKNOW_CMD       = 0;
    LOGIN            = 110000; //登录
    REQUEST_AUTHCODE = 110001; //发送验证码
    VERIFY_AUTHCODE  = 110002; //校验验证码
    KICKOUT          = 110003; //踢下线
}

enum Error_Code {
    SUCCESS = 0;
    ERROR_PASSWORD_INVALID   = 110000;
    ERROR_PASSWORD_NOT_EQUAL = 110001;
    ERROR_MEMBER_EXIST       = 110002;
    ERROR_AUTHCODE_INVALID   = 110003;
    ERROR_PHONE_INVALID      = 110004;
    ERROR_AUTHCODE_EXPIRED   = 110005;
    ERROR_MEMBER_NOT_FOUND   = 110006;
    ERROR_KICKOUT            = 110007;
    ERROR_BADREQUEST         = 110008;
}

message Login_Request {
    required string phone    = 1;
    required int32  authcode = 2;
    required string uuid     = 3;
}

message Login_Response {
    optional string phone  = 1;
    optional string secret = 2;
    optional string uuid   = 3;
    required Error_Code error_code = 4;
}

message Kickout_Response {
    required Error_Code error_code = 1;
}

message Verify_Authcode_Request {
    required string phone    = 1;
    required int32  authcode = 2;
}

message Verify_Authcode_Response {
    required Error_Code error_code = 1;
}

message Request_Authcode_Request {
    required string phone = 1;
}

message Request_Authcode_Response {
    optional int32 authcode        = 1;
    required Error_Code error_code = 2;
}
